bleak/backends/bumble/client.py

```diff
@@ -0,0 +1,322 @@
+        return True
+
+    async def unpair(self) -> bool:
+        """Unpair with the peripheral."""
+        warnings.warn(
+            "Unpairing is seemingly unavailable in the Bumble API at the moment."
+        )
+        return False
+
+    @property
+    def is_connected(self) -> bool:
+        """Check connection status between this client and the server.
+
+        Returns:
+            Boolean representing connection status.
+
+        """
+        return self._connection.connected
+
+    async def get_services(self, **kwargs) -> BleakGATTServiceCollection:
+        """Get all services registered for this GATT server.
+
+        Returns:
+           A :py:class:`bleak.backends.service.BleakGATTServiceCollection` with this device's services tree.
+
+        """
+        if not self._connection.connected:
+            raise BleakError("Not connected")
+
+        if self.services is not None:
+            return self.services
+
+        new_services = BleakGATTServiceCollection()
+        self._peer = Peer(self._connection.handle)
+        await self._peer.discover_services()
+        for service in self._peer.services:
+            new_services.add_service(BleakGATTServiceBumble(service))
+            await service.discover_characteristics()
+            for characteristic in service.characteristics:
+                await characteristic.discover_descriptors()
+                new_services.add_characteristic(
+                    BleakGATTCharacteristicBumble(
+                        characteristic, lambda: self.mtu_size - 3, service
+                    )
+                )
+                for descr in characteristic.descriptors:
+                    new_services.add_descriptor(
+                        BleakGATTDescriptorBumble(descr, characteristic)
+                    )
+
+        return new_services
+
+    async def read_gatt_char(
+        self,
+        char_specifier: Union[BleakGATTCharacteristic, int, str, uuid.UUID],
+        **kwargs,
+    ) -> bytearray:
+        """Perform read operation on the specified GATT characteristic.
+
+        Args:
+            char_specifier (BleakGATTCharacteristic, int, str or UUID): The characteristic to read from,
+                specified by either integer handle, UUID or directly by the
+                BleakGATTCharacteristic object representing it.
+
+        Returns:
+            (bytearray) The read data.
+
+        """
+        if not isinstance(char_specifier, BleakGATTCharacteristic):
+            characteristic = self.services.get_characteristic(char_specifier)
+            if not characteristic:
+                raise BleakCharacteristicNotFoundError(char_specifier)
+        else:
+            characteristic = char_specifier
+
+        value = await self._peer.read_characteristics_by_uuid(characteristic.obj.uuid)
+        value = bytearray(value[0])
+        logger.debug(f"Read Characteristic {characteristic.uuid} : {value}")
+        return value
```

Comment:
So, the peer returns a list of bytes, then we extract the first one and return it? What is in the rest of the list?

There are type errors here because value is mutated. Best to use a different variable name for the data that's extracted from the list of bytes.